src/main/java/org/entando/plugins/pda/pam/service/task/KieTaskService.java

```diff
@@ -1,18 +1,72 @@
 package org.entando.plugins.pda.pam.service.task;
 
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
 import org.entando.plugins.pda.core.engine.Connection;
 import org.entando.plugins.pda.core.model.Task;
 import org.entando.plugins.pda.core.service.task.TaskService;
+import org.entando.plugins.pda.pam.service.task.model.KieProcessVariablesResponse;
+import org.entando.plugins.pda.pam.service.task.model.KieTask;
+import org.entando.plugins.pda.pam.service.task.model.KieTasksResponse;
 import org.entando.web.request.PagedListRequest;
 import org.entando.web.response.PagedMetadata;
 import org.entando.web.response.PagedRestResponse;
+import org.springframework.boot.web.client.RestTemplateBuilder;
 import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.*;
+import java.util.stream.Collectors;
 
 @Service
+@RequiredArgsConstructor
 public class KieTaskService implements TaskService {
 
+    @NonNull
+    RestTemplateBuilder restTemplateBuilder;
+
+    public static final String TASK_LIST_URL = "/services/rest/server/queries/tasks/instances/pot-owners";
+    public static final String TASK_PROPERTIES_URL = "/services/rest/server/queries/processes/instances/{pInstanceId}/variables/instances";
+
     public PagedRestResponse<Task> list(Connection connection, PagedListRequest request) {
-        return new PagedRestResponse<>(new PagedMetadata<>(request, 0));
+        RestTemplate restTemplate = restTemplateBuilder
+                .basicAuthorization(connection.getUsername(), connection.getPassword())
+                .build();
+
+        KieTasksResponse response = restTemplate.getForObject(
+                connection.getUrl() + TASK_LIST_URL, KieTasksResponse.class);
+
+        List<Task> result = Optional.ofNullable(response.getTasks())
+                .orElse(Collections.emptyList())
+                .stream()
+                .map(this::fromDto)
+                .peek(t -> {
```

Comment:
Some general notes

1. This needs to page correctly from PAM (they have paging parameters)
2. We need to be able to pass the logged in user through here in two different ways. One is where we have the credentials of the user and pass those through to log in. The other is to pass the username as a parameter (this depends on a config setting in pam `<property name="org.kie.server.bypass.auth.user" value="true"/>`)

Use case 1: Fetching the task list for rendering. In which case we need to fetch all of the displayed values
Use case 2: Fetching process variables for display on a task table or details in use case 1
Use case 3: Fetching task variables for display on a task table or details in use case 1
Use case 4: Fetching possible values for widget configuration so that end user can pick the columns they want to see.

In the first 3 cases all of the tasks will need all of the values for display.

There are a few things going on that we'll have to figure out the best way to support. 

I think we'll need to make some of these calls configurable  for performance purposes. The old code fetches everything every time and it will bog down on a large task list. For example process variables could be skipped if they aren't selected for display or even just never fetched ever if they aren't using process variables. The same is true for task variables.

We'll also need some ability to unroll Map<> attached to tasks. A common pattern is to jam properties on tasks into a Map. I can give you an example of this